pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -369,3 +369,54 @@ func avgHistograms(head, tail []promql.HPoint) (*histogram.FloatHistogram, error
 
 	return avgSoFar, nil
 }
+
+var Changes = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction: DropSeriesName,
+	StepFunc:               changes,
+}
+
+func changes(step types.RangeVectorStepData, _ float64, _ types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
+	fHead, fTail := step.Floats.UnsafePoints(step.RangeEnd)
+
+	haveFloats := len(fHead) > 0 || len(fTail) > 0
+
+	if !haveFloats {
+		// PromQL engine doesn't support histogram for `changes` function yet,
```

Comment:
[nit] We use the term "Prometheus' engine" to refer to it - "PromQL engine" is ambiguous because MQE is a PromQL engine too.
```suggestion
		// Prometheus' engine doesn't support histogram for `changes` function yet,
```

Might also be good to add a link to the Prometheus' issue tracking adding support for native histograms in `changes`, if there is one.

Reply:
I tried to find but it seems there are no issue tracking for that.

Reply:
Charles is likely referring to this one: https://github.com/prometheus/prometheus/issues/13934

Reply:
That issue didn't mention changes. This is the comment that mentioned the need to add support ([code link](https://github.com/prometheus/prometheus/blob/7c7116fea8343795cae6da42960cacd0207a2af8/promql/functions.go#L1444)).